packages/core/src/RenderingEngine/helpers/volumeNewImageEventDispatcher.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  getImageSliceDataForVolumeViewport,
+  triggerEvent,
+} from '../../utilities';
+import { EventTypes, IImageData } from '../../types';
+import { Events } from '../../enums';
+import { getRenderingEngine } from '../getRenderingEngine';
+import VolumeViewport from '../VolumeViewport';
+
+// Keeping track of previous imageIndex for each viewportId
+type VolumeImageState = Record<string, number>;
+
+const state: VolumeImageState = {};
+
+/**
+ * It captures the camera modified event and with the camera focal point and viewPlaneNomad
+ * it calculates the image index in the view direction. Finally it triggers
+ * a VOLUME_NEW_IMAGE event with the image index.
+ *
+ * @internal
+ *
+ * @param cameraEvent - The camera modified event
+ * @param viewportImageData - The image data of the viewport
+ */
+function volumeNewImageEventDispatcher(
+  cameraEvent: EventTypes.CameraModifiedEvent,
+  viewportImageData: IImageData
+): void {
+  const { renderingEngineId, viewportId } = cameraEvent.detail;
+  const renderingEngine = getRenderingEngine(renderingEngineId);
+  const viewport = renderingEngine.getViewport(viewportId);
+
+  if (!(viewport instanceof VolumeViewport)) {
+    return;
+  }
+
+  if (!state[viewport.id]) {
+    state[viewport.id] = 0;
```

Comment:
We probably need to clean up parts of this state when elements with that volume ID are disabled (i.e. listen to element disabled). Otherwise if the same viewportId is re-used you may hit weird edge cases where an event should have fired as the index from the old viewport was the same.